app/javascript/controllers/refine/inline-advanced-modal-controller.js

```diff
@@ -49,16 +77,42 @@ export default class extends Controller {
     return this.categoryListItemTargets.find(item => item.dataset.categoryListItemValue === categoryName)
   }
 
+  clearSelection() {
+    this.debounceScrollHighlights();
+    this.categoryShortcutItemTargets.forEach((item) => {
+      item.classList.remove('active')
+    })
+  }
+
+  clearSearch() {
+    if (this.searchBarInputTarget.querySelector("input")) {
+      this.searchBarInputTarget.querySelector("input").value = ''
+    }
+  }
+
   scrollToCategory(event) {
     const categoryName = event.target.dataset.inlineAdvancedModalValue
     const categoryElement = this.findCategoryElementByName(categoryName)
+    this.shouldHighlightCategories = false;
     if(categoryElement) {
       categoryElement.scrollIntoView({
         behavior: "smooth",
         block: "start",
         inline: "nearest"
       })
     }
+    setTimeout(() => {
+      this.shouldHighlightCategories = true;
+      this.highlightCategory(categoryName, true);
+    }, 750);
+  }
```

Comment:
Should we only set `shouldHighlightCategories` and `setTimeout` if the `categoryElement` exists?

```suggestion
    if(categoryElement) {
      this.shouldHighlightCategories = false;
      categoryElement.scrollIntoView({
        behavior: "smooth",
        block: "start",
        inline: "nearest"
      })
      setTimeout(() => {
        this.shouldHighlightCategories = true;
        this.highlightCategory(categoryName, true);
      }, 750);
    }
  }
```